python/lsst/ap/verify/pipeline_driver.py

```diff
@@ -225,6 +220,53 @@ def runApPipe(metricsJob, workspace, parsedCmdLine):
         _updateMetrics(pipeline.getFullMetadata(), metricsJob)
 
 
+def _getDataRefs(butler, datasetType, dataId):
+    """Expand and resolve a partial data reference.
+
+    Parameters
+    ----------
+    workspace : `lsst.ap.verify.workspace.Workspace`
+        The abstract location containing input and output repositories.
+    datasetType : `str`
+        The type of data reference to return.
+    dataId : `dict` from `str` to any
+        Butler identifier naming the data to be processed by the underlying
+        task(s).
+
+    Returns
+    -------
+    dataRefs : iterable of `lsst.daf.persistence.ButlerDataRef`
+        All data references compatible with ``dataId``. The references shall be fully expanded.
+    """
+    refSuperset = butler.subset(datasetType=datasetType, dataId=dataId)
+    return [ref for ref in refSuperset if _dataExists(butler=butler, dataRef=ref)]
```

Comment:
Ah, yes, this bit of code. I wonder if we shouldn't lift this up into the butler itself, since I have my own versions of it floating around in jointcal (e.g. https://github.com/lsst/jointcal/blob/master/bin.src/plot_jointcal_results.py#L20 ), and I bet others do to.

Reply:
TBH, this is a near-duplicate of some code in [`pipe.base.ArgumentParser`](https://github.com/lsst/pipe_base/blob/master/python/lsst/pipe/base/argumentParser.py#L162-L166). I considered factoring it out, but `pipe_base` isn't really a natural place for this functionality. Your suggestion is better, but we'd have to *create* a Butler utilities module first.

Reply:
It figures that a couple years after killing off `daf_butlerUtils`, I end up resurrecting it...

Why can't it just live in the butler itself though? `butler.dataSetReallyExistsIPromise()`?

Reply:
To me it seems like secondary functionality (beware cluttering the API with convenience methods, etc.).

Also, IIRC Butler 2 was specifically designed to not worry about existence checks, as an optimization or something (plus it can be useful to have references to data that don't exist at the time the reference is constructed).

Reply:
Thanks for moving it into daf_persistence.